airflow/providers/microsoft/azure/hooks/powerbi.py

```diff
@@ -0,0 +1,323 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import time
+from enum import Enum
+from typing import Any, Callable
+
+import requests
+from azure.identity import ClientSecretCredential
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class PowerBIDatasetRefreshFields(Enum):
+    """Power BI refresh dataset details."""
+
+    REQUEST_ID = "request_id"
+    STATUS = "status"
+    END_TIME = "end_time"
+    ERROR = "error"
+
+
+class PowerBIDatasetRefreshStatus:
+    """Power BI refresh dataset statuses."""
+
+    # If the completion state is unknown or a refresh is in progress.
+    IN_PROGRESS = "In Progress"
+    FAILED = "Failed"
+    COMPLETED = "Completed"
+    DISABLED = "Disabled"
+
+    TERMINAL_STATUSES = {FAILED, COMPLETED}
+
+
+class PowerBIDatasetRefreshException(AirflowException):
+    """An exception that indicates a dataset refresh failed to complete."""
+
+
+class PowerBIHook(BaseHook):
+    """
+    A hook to interact with Power BI.
+
+    :param powerbi_conn_id: Airflow Connection ID that contains the connection
+        information for the Power BI account used for authentication.
+    """
+
+    conn_type: str = "powerbi"
+    conn_name_attr: str = "powerbi_conn_id"
+    default_conn_name: str = "powerbi_default"
+    hook_name: str = "Power BI"
+
+    @classmethod
+    def get_connection_form_widgets(cls) -> dict[str, Any]:
+        """Return connection widgets to add to connection form."""
+        from flask_appbuilder.fieldwidgets import BS3TextFieldWidget
+        from flask_babel import lazy_gettext
+        from wtforms import StringField
+
+        return {
+            "tenant_id": StringField(lazy_gettext("Tenant ID"), widget=BS3TextFieldWidget()),
+        }
+
+    @classmethod
+    def get_ui_field_behaviour(cls) -> dict[str, Any]:
+        """Return custom field behaviour."""
+        return {
+            "hidden_fields": ["schema", "port", "host", "extra"],
+            "relabeling": {
+                "login": "Client ID",
+                "password": "Secret",
+            },
+        }
+
+    def __init__(
+        self,
+        *,
+        powerbi_conn_id: str = default_conn_name,
+    ):
+        self.conn_id = powerbi_conn_id
+        self._api_version = "v1.0"
+        self._base_url = "https://api.powerbi.com"
+        super().__init__()
+
+    def refresh_dataset(self, dataset_id: str, group_id: str) -> str:
+        """
+        Triggers a refresh for the specified dataset from the given group id.
+
+        :param dataset_id: The dataset id.
+        :param group_id: The workspace id.
+
+        :return: Request id of the dataset refresh request.
+        """
+        url = f"{self._base_url}/{self._api_version}/myorg"
+
+        # add the group id if it is specified
+        url += f"/groups/{group_id}"
```

Comment:
When using the KiotaRequestAdapterHook, you will see you can defined path_parameters, which allows you to achieve the same, but also take care of url encoding those path parameters and already taken care for you by the Kiota library